### PR TITLE
Support Material 3 in bottom sheet

### LINKED ISSUE #111448: Update `BottomSheet` to support Material 3
As part of the [Material 3 specification](https://m3.material.io/components/bottom-sheets/overview) there are updates to bottom sheet's  visuals:

unnamed-1

We should update the `BottomSheet` widget to support this Material 3 look and tokens.

### PR DESCRIPTION
**Pull request by hannah-hyj**
issue: #111448, #111688

fixes https://github.com/flutter/flutter/issues/111448

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Nice work @hangyujin. Just have a few suggestions and a question about the M2 defaults. Thx.

File: examples/api/lib/material/bottom_sheet/show_modal_bottom_sheet.1.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/// Flutter code sample for [showModalBottomSheet].
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
```
[the remark this answers is not in the document]
Comment (hangyujin) [reply examples/api/lib/material/bottom_sheet/show_modal_bottom_sheet.1.dart]: Done

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 2]
```diff
@@ -806,3 +810,34 @@ PersistentBottomSheetController<T> showBottomSheet<T>({
...
+
+// Token database version: v0_132
+
+// Generated version v0_132
+class _BottomSheetDefaultsM3 extends BottomSheetThemeData {
+   const _BottomSheetDefaultsM3(this.context)
+    : super(
+      elevation: 1.0,
+      modalElevation: 1.0,
+      shape: const RoundedRectangleBorder(borderRadius: BorderRadius.vertical(top: Radius.circular(28.0),)),
```
[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/bottom_sheet.dart]: Done

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 3]
```diff
@@ -526,10 +529,11 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
       child: Builder(
         builder: (BuildContext context) {
           final BottomSheetThemeData sheetTheme = Theme.of(context).bottomSheetTheme;
+          final BottomSheetThemeData defaults = Theme.of(context).useMaterial3 ? _BottomSheetDefaultsM3(context) : const BottomSheetThemeData();
```
[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/bottom_sheet.dart:532]: Yeah I didn't find any default value for M2

**Review by darrenaustin — APPROVED** [review]
LGTM.

**Pull request merged**